core/src/main/java/org/apache/carbondata/core/datamap/dev/DataMapFactory.java

```diff
@@ -111,9 +111,11 @@ public abstract DataMapBuilder createBuilder(Segment segment, String shardName,
   public abstract void fireEvent(Event event);
 
   /**
-   * Clears datamap of the segment
+   * Clear all datamaps for a segment from memory
    */
-  public abstract void clear(Segment segment);
+  public void clear(String segmentNo) {
```

Comment:
Unify the names, if it is number, it should be integer, if it is id, the name should be segmentId

Reply:
done